Framework/Crystal/src/ConvertPeaksWorkspace.cpp

```diff
@@ -61,19 +61,18 @@ std::map<std::string, std::string> ConvertPeaksWorkspace::validateInputs() {
   PeaksWorkspace_sptr pws = std::dynamic_pointer_cast<PeaksWorkspace>(ipws);
   LeanElasticPeaksWorkspace_sptr lpws = std::dynamic_pointer_cast<LeanElasticPeaksWorkspace>(ipws);
 
-  // case I: missing instrument when converting to PeaksWorkspace
   if (lpws && !pws) {
+    // case I: missing instrument when converting to PeaksWorkspace
     if (getPointerToProperty("InstrumentWorkspace")->isDefault()) {
       issues["InstrumentWorkspace"] = "Need a PeaksWorkspace with proper instrument attached to assist conversion.";
     }
-  }
-
-  // case II: instrument cannot be found within donor workspace
-  if (lpws && !pws) {
-    Workspace_sptr ws = getProperty("InstrumentWorkspace");
-    ExperimentInfo_sptr inputExperimentInfo = std::dynamic_pointer_cast<ExperimentInfo>(ws);
-    if (!inputExperimentInfo) {
-      issues["InstrumentWorkspace"] = "Invalid instrument found in donor workspace.";
+    // case II: instrument cannot be found within donor workspace
+    else {
```

Comment:
I don't think this is equivalent, as `(lpws && pws)` would previously skip this branch

Reply:
Previously there were two checks for `(lpws && !pws)`, which is what cppcheck complained about. The refactor combines both into one. You're right that the logic isn't completely equivalent but I checked the new logic with the original author @KedoKudo and they were happy with the changes.